### ISSUE TITLE
Potential refutation soundness issue on UF formula

### ISSUE THREAD

**Issue by rainoftime** [SRI-CSL/yices2#315]
Hi, for this formula, yices a14eced
~~~~
z3 xx.smt2
sat
yices_smt2 xx.smt2                                                                                 
unsat
$ cat xx.smt2
(set-logic UF)
(declare-sort S0 0)
(assert (not (exists ((q3 S0) (q4 S0) (q5 Bool) (q6 Bool) (q7 S0) (q8 Bool)) (=> (= q3 q4 q7) false))))
(check-sat)
~~~~

**Comment by dddejan** [comment]
Simplified problem 
```smt2
(set-logic UF)
(declare-sort S0 0)
(assert (forall ((x S0) (y S0)) (= x y)))
(check-sat)
(get-model)
```

Output 

```console
$ z3 issue315.smt2; ./yices_smt2 issue315.smt2;
sat
(model 
  ;; universe for S0:
  ;;   S0!val!0 
  ;; -----------
  ;; definitions for universe elements:
  (declare-fun S0!val!0 () S0)
  ;; cardinality constraint:
  (forall ((x S0)) (= x S0!val!0))
  ;; -----------
)
unsat
(error "the context is unsatisfiable")
```

@aman-goel can you take a look

**Comment by aman-goel** [comment]
Resolved by db47e782d29ca85123c85d4b7e8c5654ef678dd3 . Thanks!
